driver/csiplugin/controllerserver.go

```diff
@@ -702,14 +719,50 @@ func (cs *ScaleControllerServer) CreateVolume(ctx context.Context, req *csi.Crea
 
 	}
 
+	isValidPvcFromSnapshot := false
 	if isSnapSource {
-		err = cs.validateSnapId(ctx, scaleVol, &snapIdMembers, scaleVol, primaryClusterID)
+		isValidPvcFromSnapshot, err = cs.validateSnapId(ctx, scaleVol, &snapIdMembers, scaleVol, primaryClusterID, isShallowCopyVolume)
 		if err != nil {
 			klog.Errorf("[%s] volume:[%v] - Error in source snapshot validation [%v]", loggerId, volName, err)
 			return nil, err
 		}
+
+		if !isValidPvcFromSnapshot{
+			return nil, status.Error(codes.Internal, "CreateVolume ValidateShallowCopyVolume failed")
```

Comment:
it would a good idea, if we also print a error log message here with volume details.?